packages/firestore/src/api/database.ts

```diff
@@ -635,7 +635,7 @@ export class WriteBatch implements PublicWriteBatch {
   ): WriteBatch;
   update(
     documentRef: PublicDocumentReference<unknown>,
-    fieldOrUpdateData: string | ExternalFieldPath | UpdateData,
+    fieldOrUpdateData: string | ExternalFieldPath | PublicUpdateData,
```

Comment:
Note that this will get cleaned up by https://github.com/firebase/firebase-js-sdk/pull/4038 (which uses PublicFieldPath here)